### PR TITLE
feat: Remove copy & paste from context menu on desktop

### LINKED ISSUE #2868: Remove copy/paste items from contextmenu on desktop
We should remove contextmenu `copy` & `paste` actions for non-mobile screen sizes because it's currently impossible to determine in cross-browser fashion whether a `paste` action will result in anything. As such, we should remove `copy` as well.

The reasoning is, that those actions' keyboard shortcuts are well-known enough to be ok. See discussion below.

### OLD ↓

~~Specifically, we should maybe disable it only for browsers that don't support `clipboard.readText()`, and whose in-app clipboard is empty.~~

~~image image~~

~~Just a thought. It may not be a good idea, because `cmd-v` could still result in a successful paste, and this may dissuade users from trying. At the same time, clicking on `paste` when it's not doing anything is a bad UX as well.~~

### PR DESCRIPTION
**Pull request by h7y**
Fixes #2868

### CODE REVIEW

**Comment by dwelle** [comment]
I think you should move the helper over to `is-mobile.tsx`, because `utils.ts` are used by packages which do use that sass loader which allows importing scss files.

I'd also factor out this part and reuse in both places:

https://github.com/excalidraw/excalidraw/blob/ed0bec41dcf0ca84048b5dc3aeac48c0310266a3/src/is-mobile.tsx#L13-L19

File: src/components/App.tsx [hunk 1]
```diff
@@ -3672,17 +3675,18 @@ class App extends React.Component<ExcalidrawProps, AppState> {
     ContextMenu.push({
       options: [
         actionCut,
```
Comment (dwelle) [inline src/components/App.tsx:3677]: Where we're at it, let's remove the `cut` action as well (and the separator).

**Comment by h7y** [comment]
Just noticed this. I'm on it!

**Comment by h7y** [comment]
> 
> 
> I think you should move the helper over to `is-mobile.tsx`, because `utils.ts` are used by packages which do use that sass loader which allows importing scss files.
> 
> I'd also factor out this part and reuse in both places:
> 
> https://github.com/excalidraw/excalidraw/blob/ed0bec41dcf0ca84048b5dc3aeac48c0310266a3/src/is-mobile.tsx#L13-L19

I've moved the helper to ```is-mobile.tsx```. I'm having trouble understanding the factoring out part. Should I replicate the same logic in the helper?

**Comment by dwelle** [comment]
I meant something like factoring this part out:

```
window.matchMedia 
   ? window.matchMedia(variables.isMobileQuery) 
   : (({ 
       matches: false, 
       addListener: () => {}, 
       removeListener: () => {}, 
     } as any) as MediaQueryList);
```

into a helper called `getIsMobileMatcher()`

which you'd then call in both of those places where you need it.

File: src/components/App.tsx [hunk 2]
```diff
@@ -3630,7 +3630,7 @@ class App extends React.Component<ExcalidrawProps, AppState> {
 
     const elements = this.scene.getElements();
     const element = this.getElementAtPosition(x, y);
-    const isMobile = getWhetherMobile();
+    const isMobile = getIsMobileMatcher().matches;
```
Comment (dwelle) [inline src/components/App.tsx:3633]: But we still can (should) export `const isMobile => getIsMobileMatcher().matches` instead.

**Comment by dwelle** [comment]
My suggestion above was incorrect, it should have been a function (well what I suggested wasn't even syntactically correct). Fixed.

**Comment by dwelle** [comment]
Made into `feat` coz it's quite radical change I think.

**Review by dwelle — APPROVED** [review]

**Comment by h7y** [comment]
Perfect!

**Comment by dwelle** [comment]
Thanks! :heart:

**Pull request merged**